### PR TITLE
Foundation API Conformance Update: Miscellaneous

### PR DESCRIPTION
**Pull request by itaiferber**
#### `NSBinarySearchOptions`
* `FirstEqual` → `firstEqual`
* `LastEqual` → `lastEqual`
* `InsertionIndex` → `insertionIndex`

#### `Cache`
* `Cache` → `NSCache`
* Make `NSCache` generic for type safety

#### `FileHandle`
* `func` → `var`
* `public` → `open`
* `String` → `RunLoopMode`

#### `NSExpression`
* `AnyObject` → `Any`
* `expressionValueWithObject(_ object: AnyObject?, context: NSMutableDictionary?)` → `expressionValue(with object: AnyObject?, context: NSMutableDictionary?)`

#### `Predicate`
* `Predicate` → `NSPredicate`
* Add missing `public init(format predicateFormat: String, arguments argList: CVaListPointer) { NSUnimplemented() }` and `public convenience init(format predicateFormat: String, _ args: CVarArg...) { NSUnimplemented() }`
* Update `filter`/`filtered` extensions for collection classes

#### `ComparisonPredicate`
* `ComparisonPredicate` → `NSComparisonPredicate`

#### `CompoundPredicate`
* `CompoundPredicate` → `NSCompoundPredicate`
* `public` → `open`

#### `UserDefaults`
* `func standardUserDefaults() -> UserDefaults` → `var standard: UserDefaults`
* Method name changes

#### `NumberFormatter`
* `AnyObject` → `Any`
* enum name changes
* Uncomment out `NSAttributedString` methods

#### `[NS]PersonNameComponents`
* Integrate work from overlay
* Update bridging to new model

#### `PersonNameComponentsFormatter`
* `phonetic` → `isPhonetic`

#### `MeasurementFormatter`
* Add missing `public func string<UnitType : Unit>(from measurement: Measurement<UnitType>) -> String { NSUnimplemented() }`

#### `FileManager`
* Method naming changes
* `String` → `FileAttributeKey`/`FileAttributeType` structs
* `NSFileManagerItemReplacementOptions` → `NSFileManager.ItemReplacementOptions`
* `AnyObject` → `Any`

### CODE REVIEW

**Comment by itaiferber** [comment]
@swift-ci Please test

**Comment by itaiferber** [comment]
@swift-ci Please test

File: Foundation/NSFileManager.swift [hunk 1]
```diff
@@ -576,7 +576,7 @@ open class FileManager : NSObject {
     
         If you wish to only receive the URLs and no other attributes, then pass '0' for 'options' and an empty NSArray ('[NSArray array]') for 'keys'. If you wish to have the property caches of the vended URLs pre-populated with a default set of attributes, then pass '0' for 'options' and 'nil' for 'keys'.
      */
-    open func enumerator(at url: URL, includingPropertiesForKeys keys: [URLResourceKey]?, options mask: DirectoryEnumerationOptions = [], errorHandler handler: (@escaping (URL, Error) -> Bool)? = nil) -> DirectoryEnumerator? {
+    open func enumerator(at url: URL, includingPropertiesForKeys keys: [URLResourceKey]?, options mask: DirectoryEnumerationOptions = [], errorHandler handler: ((URL, Error) -> Bool)? = nil) -> DirectoryEnumerator? {
```
Comment (phausler) [inline Foundation/NSFileManager.swift]: @jrose-apple This is a pretty gnarly error; these should be markable as `@escaping`. Do you know who would be the best to look at this?

Comment (jrose-apple) [inline Foundation/NSFileManager.swift]: I've been talking about it with @milseman and @slavapestov. Right now it *is* treated as `@escaping`, so it's not *broken;* a function type used as a generic argument is assumed to be used in arbitrary ways inside the body of the type, and that includes Optional. But we probably want to special-case Optional at some point, and then we'll have to be careful to put this back.

Comment (itaiferber) [inline Foundation/NSFileManager.swift]: Thanks for the clarification! I'll add a comment in there to explain.

Comment (jrose-apple) [inline Foundation/NSFileManager.swift]: (Good catch, Philippe.)

**Comment by itaiferber** [comment]
Third time's the charm?
@swift-ci Please test

**Comment by itaiferber** [comment]
Okay, expected failure. Working on integration on rad

**Comment by itaiferber** [comment]
Merging along with https://github.com/apple/swift-corelibs-xctest/pull/158

**Comment by itaiferber** [comment]
@swift-ci Please test

**Comment by itaiferber** [comment]
(Tested manually as well so this should work)

**Pull request merged**